### PR TITLE
Add TypeScript types

### LINKED ISSUE #15: Typings?
Can a typings file be generated for this project to support intellisense & TypeScript?

**Example:**

```ts
declare module 'geojson2h3' {
    import { Feature } from 'geojson';

    export function featureToH3Set(feature: Feature, resolution: number): string[];
    export function h3ToFeature(hexAddress: string, properties: Record<string, string | number | boolean>): Feature;
    export function h3SetToFeature(hexagons: string[], properties: Record<string, string | number | boolean>): Feature;
    export function h3SetToMultiPolygonFeature(hexagons: string[], properties: Record<string, string | number | boolean>): Feature;
    export function h3SetToFeatureCollection(hexagons: string[], getProperties: (hexAddress: string) => Record<string, string | number | boolean>): string[];
}
```

The example are what I assume the typings might be, maintainers will probably have a better idea for the `properties` args at least.

### PR DESCRIPTION
**Pull request by nrabinowitz**
- Adds TS type definitions
- Cleans up tests for `h3-js@3.7.1`
- Upgrades dev dependencies in `yarn.lock`

Fixes #15

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,10 @@
 language: node_js
 node_js:
-  - '6'
+  - '12'
```
Comment (dfellis) [inline .travis.yml:3]: Maybe add `- 14` too? It's the current LTS.

File: test/geojson2h3.spec.js [hunk 2]
```diff
@@ -134,7 +134,7 @@ function assertSymmetrical(assert, feature, hexagons) {
     );
     assertEqualFeatures(
         assert,
-        h3SetToFeature(featureToH3Set(feature, DEFAULT_RES)),
+        h3SetToFeature(featureToH3Set(feature, DEFAULT_RES).sort()),
```
Comment (dfellis) [inline test/geojson2h3.spec.js:137]: Why this change?

File: test/geojson2h3.spec.js [hunk 3]
```diff
@@ -274,8 +274,8 @@ test('featureToH3Set - MultiPolygon, contiguous', assert => {
     ];
 
     assert.deepEqual(
-        featureToH3Set(feature, DEFAULT_RES),
-        hexagons,
+        featureToH3Set(feature, DEFAULT_RES).sort(),
```
Comment (dfellis) [inline test/geojson2h3.spec.js:277]: Same here

**Review by dfellis — APPROVED** [review]
Minor comments and questions

File: test/geojson2h3.spec.js (hunk 2 above)
Comment (nrabinowitz) [reply test/geojson2h3.spec.js:137]: The test was a little brittle. `h3-js@3.7.1` changes the output order of `polyfill` results, and this test depended on the order. Sorting everything normalizes.

**Review by matthiasfeist — APPROVED** [review]
thanks for the quick PR :)

File: .travis.yml (hunk 1 above)
Comment (isaacbrodsky) [reply .travis.yml:4]: It looks like we should move CI to Github Actions as with the other H3 repos.

File: types.d.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+     * fall within the feature will be included. Note that conversion from GeoJSON
+     * is lossy; the resulting hexagon set only approximately describes the original
+     * shape, at a level of precision determined by the hexagon resolution.
+     */
+    export function featureToH3Set(feature: Feature, resolution: number): string[];
+
+    /**
+     * Convert a single H3 hexagon to a GeoJSON `Polygon` feature
+     */
+    export function h3ToFeature(h3Index: string, properties: GeoJsonProperties): Feature;
```
Comment (isaacbrodsky) [inline types.d.ts:15]: I think the underlying H3-JS was changed to h3Index being a union type accepting either string or int arrays. Do we want to expose that through here?

**Review by isaacbrodsky — APPROVED** [review]

File: types.d.ts (hunk 4 above)
Comment (nrabinowitz) [reply types.d.ts:15]: I'll leave this for a subsequent PR, as I need to review that there are no code changes needed. Should be fine AFAIK. There's a little concern about requiring a later version of `h3-js` (ideally this would be pretty liberal in what sibling package it can support), but that's probably fine.
